app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -68,6 +73,21 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
         val components = context.components
 
         return super.initializeUI(view)?.also {
+            val readerModeAction =
+                BrowserToolbar.ToggleButton(
+                    image = ContextCompat.getDrawable(requireContext(), R.drawable.ic_readermode)!!,
+                    imageSelected = ContextCompat.getDrawable(requireContext(), R.drawable.ic_readermode_selected)!!,
+                    contentDescription = requireContext().getString(R.string.browser_menu_read),
+                    contentDescriptionSelected = requireContext().getString(R.string.browser_menu_read_close),
+                    visible = { readerModeAvailable },
```

Comment:
Yes, the visibility lambda will work. However, we still need to make sure we invalidate actions when a change happens so they get re-drawn. See, comment below.